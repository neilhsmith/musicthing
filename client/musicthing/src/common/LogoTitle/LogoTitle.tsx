import styled from "styled-components";

import logo from "logo.svg";

/**
 * Displays the app's logo and title.
 */

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

const Image = styled.img`
  width: 4rem;
  height: 4rem;
  margin-right: 0.5rem;
`;

const Title = styled.h1`
  color: ${(props) => props.theme.colors.textActive};
  font-size: clamp(2.2rem, 1.4vw, 3.8rem);
`;

export default function LogoTitle() {
  return (
    <Wrapper>
      <Image src={logo} alt="logo" />
      <Title>MusicThing</Title>
    </Wrapper>
  );
}
