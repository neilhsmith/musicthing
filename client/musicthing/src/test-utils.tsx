import React, { ReactElement } from "react";
import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import { dark } from "styles/themes";

const AllProviders = ({ children }: any) => {
  return <ThemeProvider theme={dark}>{children}</ThemeProvider>;
};

const customRender = (ui: ReactElement, options?: any) =>
  render(ui, { wrapper: AllProviders, ...options });

export * from "@testing-library/react";

export { customRender as render, screen, dark as defaultTheme };
