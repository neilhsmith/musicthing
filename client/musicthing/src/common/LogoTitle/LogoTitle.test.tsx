import { render, screen, defaultTheme } from "test-utils";

import LogoTitle from "./LogoTitle";

describe("Component: LogoTitle", () => {
  it("matches its snapshot", () => {
    const { container } = render(<LogoTitle />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("displays the logo", () => {
    render(<LogoTitle />);

    expect(screen.getByRole("img")).toBeInTheDocument();
  });

  it("displays the title", () => {
    render(<LogoTitle />);

    expect(screen.getByRole("heading")).toHaveTextContent("MusicThing");
  });

  it("uses the theme's textActive as the title's color", () => {
    render(<LogoTitle />);

    expect(screen.getByRole("heading")).toHaveStyleRule(
      "color",
      defaultTheme.colors.textActive
    );
  });
});
