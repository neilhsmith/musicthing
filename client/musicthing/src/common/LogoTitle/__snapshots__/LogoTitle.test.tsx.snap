// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LogoTitle matches its snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 4rem;
  height: 4rem;
  margin-right: 0.5rem;
}

.c2 {
  color: #ffffff;
  font-size: clamp(2.2rem,1.4vw,3.8rem);
}

<div
  class="c0"
>
  <img
    alt="logo"
    class="c1"
    src="logo.svg"
  />
  <h1
    class="c2"
  >
    MusicThing
  </h1>
</div>
`;
